### DISCUSSION TITLE
React application shows 404 after page refresh

### DISCUSSION THREAD

**Discussion by keinsell** [vercel/community#261] [Help] [answered]
I have a simple `React.js` application with `wouter`-based routing and `Vercel` deployments that drive me into a little issues where I have no idea about the solution. It always returns `404` after refresh (of subpage).

My code looks a bit like this.

```jsx
	<>
		<Route path='/' component={Intro} />
		<Route path='/dashboard' component={Dashboard} />
	</>
```

Where on `<Intro />` I have `<Link />` to `/dashboard` which should transfer me to `<Dashboard />` page. And it does work on my local machine, in a container, and on Linux-based deployment, but it doesn't really work in `vercel` deployment, even if I tried to resolve this issue with the configuration of `vercel.json`.

```json
{
	"github": {
		"silent": true
	},
	"rewrites": [
		{
			"source": "(.*)",
			"destination": "/index.html"
		}
	]
}
```

Link to live deployment available here: pointrest-6ttd9zep8-araclx.vercel.app

> NOTE: I was also trying to use `react-router` but the same problem exists.

**Comment by samcx** [comment]
@keinsell Can you try adding the slash here → ` "source": "/(.*)",`?

**Reply by boypanjaitan16** [reply]
Hi I tried this but it's only works for top level path like `/profile` but it's not work for multilevel path like `/profile/orders`. any idea how to write `"source"` path to match all the path? thanks

**Reply by samcx** [reply]
`:match*` is used for multipath @boypanjaitan16 https://vercel.com/docs/project-configuration#project-configuration/rewrites

**Comment by keinsell** [comment, marked as answer]
I figured it out, `vercel.json` was fine but it was in the root of my repo where I was using `turborepo`. After moving it to package with `vite` everything was working fine with my config.

**Reply by Aycom366** [reply]
may I asked   where you moved the `vercel.json` file becuase am also having same issue with vite
my vercel.json is sitting at the root folder

**Reply by keinsell** [reply]
@Aycom366 Move your `vercel.json` to directory where you have application that is going to be deployed, in my case it was `apps/web/vercel.json`.

**Reply by bloodstormm** [reply]
May i ask you guys where did u find `vercel.json`? Is it at the your github repo?

**Reply by wisdom-ossai** [reply]
When you create the `vercel.json` file at the root of your folder, just add this piece of code and redeploy. This should fix it

`{
  "routes": [{ "src": "/[^.]+", "dest": "/", "status": 200 }]
}`

**Reply by Charizard13** [reply]
In Vite - React does it need to be in General Dir, inside the src, or in the public folder where static assets are? :)

**Reply by ELHart05** [reply]
how to solve the problem in generally let's say you're hosting the spa on another domain?

**Reply by nehatkkhan786** [reply]
yeah bro I am also having the same issue. Please someone clearly clear the steps to fix this issue.

**Reply by dvmhmdsd** [reply]
For all who are struggling with this issue, just put the `vercel.json` file inside the build folder that is generated by vite build.

To make it automatically put on every build rather than manually put it inside the generated build folder, put the `vercel.json` inside the `public` folder and it will be moved automatically to the generated build folder.

To make sure that the problem has been solved also create a `_redirects` file inside the `public` folder with this content:
``` /* /index.html 200 ```

This will solve the redirect issue of other platforms like `netlify`

**Reply by zishanakhtar** [reply]
@dvmhmdsd Your suggestion of moving `vercel.json` to `public` before generating the build worked for me. Thanks.

**Comment by shyamlohar** [comment]
{
    "rewrites": [{ "source": "/(.*)", "destination": "/" }]
}
 
 works for me for domain given by vercel. but issue persists for custom domain url.

**Reply by samcx** [reply]
@shyamlohar Is it possible to share with us both domains for this so we can take a closer look?

**Reply by anjantalatam** [reply]
## For React Vercel project

This resolved my issue. ✅ 

Add the below code to vercel.json in the root directory 

```
{
     "rewrites": [{ "source": "/(.*)", "destination": "/" }]
}
```

Project created with `yarn create-vite`

Deployed on Vercel: https://course-hub-theta.vercel.app/tutor

**Reply by elioenays** [reply]
It worked

**Reply by Priyam7295** [reply]
> It worked
Finally , found something useful ,it worked

**Comment by shyamlohar** [comment]


**Reply by amyegan** [reply]
I'm not able to see a 404 on the custom domain URL. Have you updated the deployment to no longer have the error?

Or maybe I'm doing something wrong. Can you share the exact steps I should take to see the error that you see?

**Reply by dcoder-x** [reply]
Have you been able to resolve the issue?

**Comment by shyamlohar** [comment]
Yes we were able to fix this issue

Thank you
Shyam Lohar

**Comment by nehatkkhan786** [comment]
https://youtu.be/9V5F-OnPsms
check out this video, it covers the fix which is mention above.
